llvm/lib/Target/AArch64/AArch64TargetTransformInfo.cpp

```diff
@@ -2150,13 +2150,42 @@ static std::optional<Instruction *> instCombineSVEInsr(InstCombiner &IC,
   return std::nullopt;
 }
 
+static std::optional<Instruction *> instCombineDMB(InstCombiner &IC,
+                                                   IntrinsicInst &II) {
+  // If this barrier is post-dominated by identical one we can remove it
+  auto *NI = II.getNextNonDebugInstruction();
+  int LookaheadThreshold = 10;
+  auto CanSkipOver = [](Instruction *I) {
+    return !I->mayReadOrWriteMemory() && !I->mayHaveSideEffects();
+  };
+  while (--LookaheadThreshold && !isa<IntrinsicInst>(NI)) {
```

Comment:
Can you fold the call below into the while statement - does that work out neater? For example:

```
  while (--LookaheadThreshold && !isa<IntrinsicInst>(NI) && CanSkipOver(NI)) {
    auto *NIBB = NI->getParent();
    ...
```

Reply:
Not knowing the exact use case I'm just wondering if there are any concerns over the looseness of bailing for any intrinsic, even something like `llvm.ctlz` which can be safely skipped over?